### ISSUE TITLE
WIP: Figure out how to model a response

### ISSUE THREAD

**Issue by hizumisen** [casualteam/open-api-generator#5]
Right now the response model is a case class where all possible results must be defined and instantiated

```scala
case class PetInsert200 (
  `application/json`: Pet ,
  `application/xml`: Pet
)
```

these models are all requested since they differ by content type and the generated logic choose which response actually return.
This solution could lead to inefficient code.

Some different model could be

Model 1:
```scala
case class PetInsert200 (
  `application/json`: () => Pet,
  `application/xml`: () => Pet
)
```

Model 2:
```scala
case class PetInsert200 (
  `application/json`: () => Future[Pet],
  `application/xml`: () => Future[Pet]
)
```

**Comment by datalek** [comment]
Solution 1 is less "smart" than solution 2. Let me explain with an example; 

Suppose an endpoint that uses the `Content-Type` to differentiate the input (`application/json;version=1` or `application/json;version=2`), and suppose to have different behaviour for each version, a.k.a. two different functions, one function handle the *version-1* branch, and the other one the *version-2* branch.
```scala
def handleVersion1(): Future[Pet] = ...
def handleVersoin2(): Future[Pet] = ...
```

Solution 1 *runs* each function all the time, because you need to *unbox* the value from the future:
```scala
case class PetInsert200 (
  `application/json;version=1`: () => Pet,
  `application/json;version=2`: () => Pet
)

for {
  v1 <- handleVersion1()
  v2 <- handleVersion2()
} yield PetInsert200(v1, v2)
```

Solution 2 is much better because it allows you to *run* only the branch you need, you don't need to *unbox* the value from the future:
```scala
case class PetInsert200 (
  `application/json;version=1`: () => Future[Pet],
  `application/json;version=2`: () => Future[Pet]
)

PetInsert200(handleVersion1, handleVersion2)
```

**Comment by hizumisen** [comment]
This issue will be addressed in #1 because it regards endpoint interface but on the response side.
